VMware LRO fixes for rotate-vcenter-password, rotate-nsxt-password, and restrict-movement

jonathanhe-msft opened the pull request:
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az vmware private-cloud rotate-vcenter-password`, `az vmware private-cloud rotate-nsxt-password`, and `az vmware vm restrict-movement`


### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).


### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

yonzhan commented:
Thank you for your contribution! We will review the pull request and get back to you soon.

cataggar commented on src/vmware/azext_vmware/azext_metadata.json:
```diff
@@ -1,4 +1,4 @@
 {
     "azext.isPreview": false,
-    "azext.minCliCoreVersion": "2.54.0"
+    "azext.minCliCoreVersion": "2.67.0"
```
Bumping the minimum AZ CLI version from 2.54 to 2.67 probably warrents bumping the miror version to `7.2.0`.

cataggar commented on src/vmware/azext_vmware/aaz/latest/vmware/addon/_create.py:
```diff
@@ -44,15 +44,15 @@ def _build_arguments_schema(cls, *args, **kwargs):
             required=True,
             fmt=AAZStrArgFormat(
-                pattern="^[-\w\._]+$",
+                pattern="^[-\\w\\._]+$",
```
Can you create a PR with just the bug fix and not all of these additional changes?

jonathanhe-msft replied:
i believe the additional changes for the regex and readonly properties are there because i used a newer version of the aaz-dev-tools to generate the client. i can't go back to the old aaz-dev-tools version since it doesn't support python 3.12. i tried regenerating with the earliest version that supports python 3.12 and still see the additional changes

zhoxing-ms commented on src/vmware/azext_vmware/aaz/latest/vmware/private_cloud/_wait.py:
```diff
@@ -198,7 +198,9 @@ def _build_schema_on_200(cls):
             )
             properties.encryption = AAZObjectType()
-            properties.endpoints = AAZObjectType()
+            properties.endpoints = AAZObjectType(
+                flags={"read_only": True},
+            )
```
I see that many properties have been marked as read-only. May I ask what is the reason?

jonathanhe-msft replied:
It's not due to swagger change as swagger has had these values since it was created. 

I believe that it's due to the python version upgrade and aaz-dev-tools upgrades. I have double checked and the properties do exist in swagger, so there could have been upgrades to code-gen tool that now capture these property flags.

jonathanhe-msft commented:
When generating the client using `aaz-dev`, I noticed that new line characters in confirmation texts are escaped with a double `\` eg `\\n`. When I run these commands, the text is formatted incorrectly (does not display as a new line, instead displays the now escaped new line  characters. I modified the text manually to revert it back to the new lines. This can be seen in the second to last commit. 

Can I get some confirmation that this is incorrectly generated and `\n` should not be escaped?

zhoxing-ms commented:
> When generating the client using aaz-dev, I noticed that new line characters in confirmation texts are escaped with a double \ eg \\n. When I run these commands, the text is formatted incorrectly (does not display as a new line, instead displays the now escaped new line characters. I modified the text manually to revert it back to the new lines. This can be seen in the second to last commit. Can I get some confirmation that this is incorrectly generated and \n should not be escaped?

@kairu-ms This issue seems to be related to `aaz-dev`~ If so, could you please take a look at this Code Gen related question?

zhoxing-ms commented:
Could you please resolve these CI issues?

kairu-ms commented:
> When generating the client using `aaz-dev`, I noticed that new line characters in confirmation texts are escaped with a double `\` eg `\\n`. When I run these commands, the text is formatted incorrectly (does not display as a new line, instead displays the now escaped new line characters. I modified the text manually to revert it back to the new lines. This can be seen in the second to last commit.
> 
> Can I get some confirmation that this is incorrectly generated and `\n` should not be escaped?

There’s no need to add \n in help. I think you can sample remove them

jonathanhe-msft commented:
> > When generating the client using `aaz-dev`, I noticed that new line characters in confirmation texts are escaped with a double `\` eg `\\n`. When I run these commands, the text is formatted incorrectly (does not display as a new line, instead displays the now escaped new line characters. I modified the text manually to revert it back to the new lines. This can be seen in the second to last commit.
> > Can I get some confirmation that this is incorrectly generated and `\n` should not be escaped?
> 
> There’s no need to add \n in help. I think you can sample remove them

We want to keep the line spacing though. It's a long message and is much more readable when split up

![{4C90BDD0-5D4D-4036-B996-3A91C7C8794E}](https://github.com/user-attachments/assets/26e72589-0e8a-4bc2-8b38-c0103ed0499e)

kairu-ms commented:
> When generating the client using `aaz-dev`, I noticed that new line characters in confirmation texts are escaped with a double `\` eg `\\n`. When I run these commands, the text is formatted incorrectly (does not display as a new line, instead displays the now escaped new line characters. I modified the text manually to revert it back to the new lines. This can be seen in the second to last commit.
> 
> Can I get some confirmation that this is incorrectly generated and `\n` should not be escaped?

necusjz commented:
> > > When generating the client using `aaz-dev`, I noticed that new line characters in confirmation texts are escaped with a double `\` eg `\\n`. When I run these commands, the text is formatted incorrectly (does not display as a new line, instead displays the now escaped new line characters. I modified the text manually to revert it back to the new lines. This can be seen in the second to last commit.
> > > Can I get some confirmation that this is incorrectly generated and `\n` should not be escaped?
> > 
> > 
> > There’s no need to add \n in help. I think you can sample remove them
> 
> We want to keep the line spacing though. It's a long message and is much more readable when split up
> 
> ![{4C90BDD0-5D4D-4036-B996-3A91C7C8794E}](https://private-user-images.githubusercontent.com/88111889/417611540-26e72589-0e8a-4bc2-8b38-c0103ed0499e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA3MTI0NjEsIm5iZiI6MTc0MDcxMjE2MSwicGF0aCI6Ii84ODExMTg4OS80MTc2MTE1NDAtMjZlNzI1ODktMGU4YS00YmMyLThiMzgtYzAxMDNlZDA0OTllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjI4VDAzMDkyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI0ZDE5YjBhOTRhMDkyODg2N2ExNmRlZDI3NTgxNTRhNzE0OTdlNDgzNGE5YTJmNzdhYjZlNzAzYzNkNzk2YTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VxjEEt_VlpdBw1uE3xVOR_XPkGYmvYmqpSt2PgIGR9Y)

@jonathanhe-msft have you tried add a new line by `\n\n`? E.g.,

![image](https://github.com/user-attachments/assets/6dea96e6-140f-4811-9657-8f3db8330e6c)
![image](https://github.com/user-attachments/assets/65306d96-389e-4dc8-989f-1bc89b20548e)

jonathanhe-msft commented:
> > > > When generating the client using `aaz-dev`, I noticed that new line characters in confirmation texts are escaped with a double `\` eg `\\n`. When I run these commands, the text is formatted incorrectly (does not display as a new line, instead displays the now escaped new line characters. I modified the text manually to revert it back to the new lines. This can be seen in the second to last commit.
> > > > Can I get some confirmation that this is incorrectly generated and `\n` should not be escaped?
> > > 
> > > 
> > > There’s no need to add \n in help. I think you can sample remove them
> > 
> > 
> > We want to keep the line spacing though. It's a long message and is much more readable when split up
> > ![{4C90BDD0-5D4D-4036-B996-3A91C7C8794E}](https://private-user-images.githubusercontent.com/88111889/417611540-26e72589-0e8a-4bc2-8b38-c0103ed0499e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA3MTI0NjEsIm5iZiI6MTc0MDcxMjE2MSwicGF0aCI6Ii84ODExMTg4OS80MTc2MTE1NDAtMjZlNzI1ODktMGU4YS00YmMyLThiMzgtYzAxMDNlZDA0OTllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjI4VDAzMDkyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI0ZDE5YjBhOTRhMDkyODg2N2ExNmRlZDI3NTgxNTRhNzE0OTdlNDgzNGE5YTJmNzdhYjZlNzAzYzNkNzk2YTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VxjEEt_VlpdBw1uE3xVOR_XPkGYmvYmqpSt2PgIGR9Y)
> 
> @jonathanhe-msft have you tried add a new line by `\n\n`? E.g.,
> 
> ![image](https://private-user-images.githubusercontent.com/12371639/417864621-6dea96e6-140f-4811-9657-8f3db8330e6c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEwNTYwNjUsIm5iZiI6MTc0MTA1NTc2NSwicGF0aCI6Ii8xMjM3MTYzOS80MTc4NjQ2MjEtNmRlYTk2ZTYtMTQwZi00ODExLTk2NTctOGYzZGI4MzMwZTZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA0VDAyMzYwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkyYWJmNTg4ZDY2NzU2N2E1ZWI0MjhiYjBiMzU4ZDA2NWEzNmVmZDBlOGQxNjIzODU5YWFhNzNmODI2N2I3ZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.GX95Pe8t78OVMKjrPXfa-jgFR30zpQkT_2MZR5zXaHA) ![image](https://private-user-images.githubusercontent.com/12371639/417864681-65306d96-389e-4dc8-989f-1bc89b20548e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEwNTYwNjUsIm5iZiI6MTc0MTA1NTc2NSwicGF0aCI6Ii8xMjM3MTYzOS80MTc4NjQ2ODEtNjUzMDZkOTYtMzg5ZS00ZGM4LTk4OWYtMWJjODliMjA1NDhlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA0VDAyMzYwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVmNjYzYWRkMjU2ZTFhMWZjM2JlMWEzNTc3NDExM2Y3Y2RhZjI2NDUwMDU5YzI2NTMxYTdkMGE2YmM2OGUzN2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jxIGbAw7_z3KrQZqrMYBJnkxl7WoyxqCUzkIxTKNZ0o)

it currently uses \n\n and looks good

LEGAL TERMS\n\nAzure VMware Solution (\"AVS\") is an Azure Service licensed to you as part of your Azure subscription and subject to the terms and conditions of the agreement under which you obtained your Azure subscription (https://azure.microsoft.com/support/legal/). The following additional terms also apply to your use of AVS:\n\nDATA RETENTION.

necusjz commented:
@jonathanhe-msft plz fix the pylint issues of R0914 and W0212 https://dev.azure.com/azclitools/public/_build/results?buildId=224185&view=logs&j=506b8f13-8269-5210-246e-a3964e53141e&t=db7247cb-300d-5b59-fd6c-6258fad20b89&l=269

necusjz commented:
@jonathanhe-msft plz provide the link of pr in Azure/aaz as well.

necusjz commented:
@jonathanhe-msft ok, the following breaking changes are as expected?

![image](https://github.com/user-attachments/assets/65ccf992-23fb-48e2-a8fe-d6e0cc28adac)

jonathanhe-msft commented:
> @jonathanhe-msft ok, the following breaking changes are as expected?
> 
> ![image](https://private-user-images.githubusercontent.com/12371639/422569614-65ccf992-23fb-48e2-a8fe-d6e0cc28adac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIyNDE3NTcsIm5iZiI6MTc0MjI0MTQ1NywicGF0aCI6Ii8xMjM3MTYzOS80MjI1Njk2MTQtNjVjY2Y5OTItMjNmYi00OGUyLWE4ZmUtZDZlMGNjMjhhZGFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE3VDE5NTczN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjOTcxYTcyMzgzM2MzY2JlZWE5YTViMzAyZjBlZDAyOTcwYjE4MTRlY2ZiNDQzYzVlYjczNmQ5NTJhMzU3MzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.17YnOZk8ML507AKQ8JIhn8YVmvl9qDAz1fBCa5emh-U)

Yes, those breaking changes are as expected. Those arguments are unused and were caught by the pylint issues. I fixed in this commit: https://github.com/Azure/azure-cli-extensions/pull/8462/commits/b626702201df4cc8ce2bcccb5b787215b0c510b6 

Added a line in `HISTORY.md` to document change

jonathanhe-msft commented:
@necusjz can I get another review on this? thanks!
